Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyBoolComboBoxCtrl.cpp

```diff
@@ -93,16 +123,53 @@ namespace AzToolsFramework
 
     void BoolPropertyComboBoxHandler::ConsumeAttribute(PropertyBoolComboBoxCtrl* GUI, AZ::u32 attrib, PropertyAttributeReader* attrValue, const char* debugName)
     {
-        AZ_UNUSED(GUI);
-        AZ_UNUSED(attrib);
-        AZ_UNUSED(attrValue);
         AZ_UNUSED(debugName);
+
+        if (attrib == AZ_CRC_CE("EditButtonVisible"))
+        {
+            bool visible;
+            if (attrValue->Read<bool>(visible))
+            {
+                GUI->GetEditButton()->setVisible(visible);
+            }
+        }
+        else if (attrib == AZ_CRC_CE("SetTrueLabel"))
+        {
+            const char* label;
+            attrValue->Read<const char*>(label);
```

Comment:
Are you sure that `<const char*>` is valid usage for `Read`. There is no other place in the code using in like that, all the other places use `<AZStd::string>`. 

Is it safe to use `<const char*>`? Is the pointer that it returns still valid?

Same for `SetFalseLabel` and `EditButtonToolTip`.